src/lib/comms/sol-coap.c

```diff
@@ -1519,19 +1528,23 @@ respond_packet(struct sol_coap_server *server, struct sol_coap_packet *req,
     /* If it isn't a request. */
     if (code & ~SOL_COAP_REQUEST_MASK) {
         bool found_reply = false;
+        bool reply_callback_result = false;
```

Comment:
Move this declaration one level down, inside the loop block. Seems that you don't need it outside the loop.